src/thor/costmatrix.cc

```diff
@@ -338,8 +338,8 @@ void CostMatrix::Initialize(
       // Allocate the adjacency list and hierarchy limits for this source.
       // Use the cost threshold to size the adjacency list.
       edgelabel_[exp_dir][i].reserve(max_reserved_labels_count_);
-      adjacency_[exp_dir][i].reuse(0, current_cost_threshold_, costing_->UnitSize(),
-                                   &edgelabel_[exp_dir][i]);
+      adjacency_[exp_dir][i].reuse(0, GetBucketRange(current_pathdist_threshold_),
```

Comment:
I'd still take the previous logic to estimate the cost range for the queue. or do what bidirectional_astar does and just make it a constant (20k currently)?